### PR TITLE
Add example of redirectional coordinator

### PR DESCRIPTION
**Pull request by d2vydov**

### CODE REVIEW

File: Example/XCoordinator/Coordinators/AboutCoordinator.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+class AboutCoordinator: RedirectionCoordinator<UserListRoute, NavigationTransition> {
+    
+    // MARK: - Init
+    
+    init<T: TransitionPerformer>(superCoordinator: T) where T.TransitionType == NavigationTransition {
+        let vc = UIViewController()
+        vc.loadViewIfNeeded()
+        vc.view.backgroundColor = .green
+        super.init(viewController: vc, superTransitionPerformer: superCoordinator, prepareTransition: nil)
+    }
```
Comment (pauljohanneskraft) [inline Example/XCoordinator/Coordinators/AboutCoordinator.swift]: I would prefer an empty AboutRoute and to override prepareTransition with `switch route {}` to show, that you should either provide a prepareTransition closure in the initializer or override the prepareTransition method.

**Comment by pauljohanneskraft** [comment]
It would also be nice, if you could remove all the other changes not related to the actual new AboutCoordinator, i.e. changes in Podfile.lock, Manifest.lock, etc

**Comment by d2vydov** [comment]
> It would also be nice, if you could remove all the other changes not related to the actual new AboutCoordinator, i.e. changes in Podfile.lock, Manifest.lock, etc

yep, fixed :)

File: Example/XCoordinator/Coordinators/AboutCoordinator.swift (hunk 1 above)
Comment (d2vydov) [reply Example/XCoordinator/Coordinators/AboutCoordinator.swift]: done :)

File: Example/XCoordinator/Coordinators/AboutCoordinator.swift [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+enum AboutRoute: Route {
+    
+}
+
+class AboutCoordinator: RedirectionCoordinator<AboutRoute, NavigationTransition> {
+    
+    // MARK: - Init
+    
+    init<T: TransitionPerformer>(superCoordinator: T) where T.TransitionType == NavigationTransition {
+        let vc = UIViewController()
```
Comment (pauljohanneskraft) [inline Example/XCoordinator/Coordinators/AboutCoordinator.swift]: I'm sorry, but I found two small issues again - please call this variable `viewController` and you do not need the call to loadViewIfNeeded

File: Example/XCoordinator/Scenes/Home/HomeViewController.swift [hunk 3]
```diff
@@ -18,6 +18,8 @@ class HomeViewController: UIViewController, BindableType {
     @IBOutlet private var logoutButton: UIButton!
     @IBOutlet private var usersButton: UIButton!
 
+    @IBOutlet var aboutButton: UIButton!
```
Comment (pauljohanneskraft) [inline Example/XCoordinator/Scenes/Home/HomeViewController.swift]: Please make the aboutButton private

File: Example/XCoordinator/Coordinators/UserListCoordinator.swift [hunk 4]
```diff
@@ -45,6 +46,9 @@ class UserListCoordinator: NavigationCoordinator<UserListRoute> {
             return registerPeek(for: source, route: .users)
         case .logout:
             return .dismiss()
+        case .about:
+            let coordinator = AboutCoordinator(superCoordinator: self)
+            return .push(coordinator.anyCoordinator)
```
Comment (pauljohanneskraft) [inline Example/XCoordinator/Coordinators/UserListCoordinator.swift]: You do not actually need to write `.anyCoordinator` here, so please just write `.push(coordinator)`. It is helpful, as it highlights the current problem, but not necessary for the Example app in general.

File: Example/XCoordinator/Coordinators/AboutCoordinator.swift (hunk 2 above)
Comment (d2vydov) [reply Example/XCoordinator/Coordinators/AboutCoordinator.swift]: I wanna change background color, yep, `loadViewIfNeeded` is required

File: Example/XCoordinator/Coordinators/UserListCoordinator.swift (hunk 4 above)
Comment (d2vydov) [reply Example/XCoordinator/Coordinators/UserListCoordinator.swift]: ok

File: Example/XCoordinator/Coordinators/AboutCoordinator.swift (hunk 2 above)
Comment (d2vydov) [reply Example/XCoordinator/Coordinators/AboutCoordinator.swift]: renamed vc -> viewController

Comment (d2vydov) [reply Example/XCoordinator/Coordinators/AboutCoordinator.swift]: you can't change background color, if the view has not loaded yet. the `loadIfNeeded` will prepare view immediately

Comment (pauljohanneskraft) [reply Example/XCoordinator/Coordinators/AboutCoordinator.swift]: But as soon as you access the view property, it automatically loads it, if needed

Comment (d2vydov) [reply Example/XCoordinator/Coordinators/AboutCoordinator.swift]: oh, you right! thank you <3

Comment (d2vydov) [reply Example/XCoordinator/Coordinators/AboutCoordinator.swift]: fixed

**Pull request merged**